model/task_lifecycle.go

```diff
@@ -1449,18 +1449,32 @@ func ClearAndResetStrandedTask(h *host.Host) error {
 		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &t.Details, false))
 	}
 
-	if t.IsPartOfDisplay() {
-		dt, err := t.GetDisplayTask()
+	return errors.Wrap(ResetTaskOrDisplayTask(t, evergreen.User, evergreen.MonitorPackage, &t.Details), "problem resetting task")
+}
+
+// ResetTaskOrDisplayTask is a wrapper for TryResetTask that handles execution and display tasks that are restarted
+// from sources separate from marking the task finished. If an execution task, attempts to restart the display task instead.
+// Marks display tasks as reset when finished and then check if it can be reset immediately.
+func ResetTaskOrDisplayTask(t *task.Task, user, origin string, detail *apimodels.TaskEndDetail) error {
```

Comment:
ResetTaskOrDisplayTask I realize is not a good name, but I can't think of a better one without completely refactoring all of the logic which is out of scope here. Let me know if you think of a better one! I wanted to move the ClearAndResetStrandedTask logic to a shared function to help with testing and code deduplication.

Reply:
I'm not against the name, it makes it clear what the function is doing.